src/Symfony/Component/Runtime/SymfonyRuntime.php

```diff
@@ -0,0 +1,205 @@
+                $arguments[] = $this->getArgument($parameter);
+            }
+        } catch (\InvalidArgumentException $e) {
+            if (!$parameter->isOptional()) {
+                throw $e;
+            }
+        }
+
+        if (!$arguments) {
+            return $app;
+        }
+
+        return static function () use ($app, $arguments) {
+            return $app(...$arguments);
+        };
+    }
+
+    public function getMain(object $app): \Closure
+    {
+        if ($app instanceof HttpKernelInterface) {
+            $request = $this->request ?? $this->request = Request::createFromGlobals();
+
+            return static function () use ($app, $request): int {
+                $response = $app->handle($request);
+                $response->send();
+
+                if ($app instanceof TerminableInterface) {
+                    $app->terminate($request, $response);
+                }
+
+                return 0;
+            };
+        }
+
+        if ($app instanceof Response) {
+            return static function () use ($app): int {
+                $app->send();
+
+                return 0;
+            };
+        }
+
+        if ($app instanceof Command) {
+            $application = $this->application ?? $this->application = new Application();
+            $main = $this->getMain($application);
+
+            return static function () use ($app, $application, $main): int {
+                $application->setName($app->getName() ?: $application->getName());
+
+                if (!$app->getName() || !$application->has($app->getName())) {
+                    $app->setName($_SERVER['argv'][0]);
+                    $application->add($app);
+                }
+
+                $application->setDefaultCommand($app->getName(), true);
+
+                return $main();
+            };
+        }
+
+        if ($app instanceof Application) {
+            if (!\in_array(\PHP_SAPI, ['cli', 'phpdbg', 'embed'], true)) {
+                echo 'Warning: The console should be invoked via the CLI version of PHP, not the '.\PHP_SAPI.' SAPI'.\PHP_EOL;
+            }
+
+            set_time_limit(0);
+            $input = $this->getInput();
+
+            return static function () use ($app, $input): int {
+                $definition = $app->getDefinition();
+
+                if (!$definition->hasOption('env')) {
+                    $definition->addOption(new InputOption('--env', '-e', InputOption::VALUE_REQUIRED, 'The Environment name.', $_SERVER['APP_ENV']));
+                }
+
+                if (!$definition->hasOption('no-debug')) {
+                    $definition->addOption(new InputOption('--no-debug', null, InputOption::VALUE_NONE, 'Switches off debug mode.'));
+                }
+
+                return $app->run($input);
+            };
+        }
+
+        throw new \LogicException(sprintf('Cannot handle return value of type "%s".', \get_class($app)));
```

Comment:
Old text is still here.  `Cannot handle application of type`

Reply:
updated